test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/InputTagHelperTest.cs

```diff
@@ -252,6 +252,304 @@ public static TheoryData MultiAttributeCheckBoxData
             Assert.Equal(expectedPostElement, output.PostElement.GetContent());
         }
 
+        [Theory]
+        [InlineData("checkbox")]
+        [InlineData("hidden")]
+        [InlineData("number")]
+        [InlineData("password")]
+        [InlineData("text")]
+        public void Process_WithEmptyForName_Throws(string inputTypeName)
+        {
+            // Arrange
+            var expectedMessage = "The name of an HTML field cannot be null or empty. Instead use methods " +
```

Comment:
My personal preference for these things is to always use the resource (I know Eilon feels differently); so much pain in maintaining these error messages.

Reply:
I'm sticking w/ @Eilon's preference.